addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -3229,14 +3229,15 @@ var SnippetsMenu = Widget.extend({
      *    false: always fobid
      */
     _insertDropzone: function ($hook, vertical, style, canBeSanitizedUnless) {
+        const skip = $hook.closest('[data-oe-sanitize="no_block"]').length;
         let forbidSanitize;
         if (canBeSanitizedUnless === 'form') {
-            forbidSanitize = $hook.closest('[data-oe-sanitize]:not([data-oe-sanitize="allow_form"])').length;
+            forbidSanitize = $hook.closest('[data-oe-sanitize]:not([data-oe-sanitize="allow_form"]):not([data-oe-sanitize="no_block"])').length;
         } else {
-            forbidSanitize = !canBeSanitizedUnless && $hook.closest('[data-oe-sanitize]').length;
+            forbidSanitize = !canBeSanitizedUnless && $hook.closest('[data-oe-sanitize]:not([data-oe-sanitize="no_block"])').length;
         }
         var $dropzone = $('<div/>', {
-            'class': 'oe_drop_zone oe_insert' + (vertical ? ' oe_vertical' : '') +
+            'class': skip ? 'd-none' : 'oe_drop_zone oe_insert' + (vertical ? ' oe_vertical' : '') +
```

Comment:
What's the point to create and insert a hidden dropzone on a field that can't receive dropped snippet?
Can't we shortcircuit and `return;`?

Reply:
I did choose not to do it in order to avoid updating all callers to this method which do chain call the insertion in the DOM. But indeed it could have been done.

Reply:
You mean that expect a jquery Element set as returned value? That's what I checked and I don't think there is a caller that actually do something with the returned value.

Reply:
:facepalm: I misread `self._insertDropzone($('<we-hook/>').insertAfter($clone)...` as `self._insertDropzone($('<we-hook/>')).insertAfter($clone)...` - and it somehow stuck like that in my understanding.